src/import.cmd.js

```diff
@@ -45,11 +55,41 @@ export default class ImportCommand extends AbstractServerCommand {
     return this;
   }
 
+  /**
+   * Ensure the UI branch the caller specified (or defaulted to) is the one being used.
+   * @param uiFolder
+   * @returns {Promise<void>}
+   */
+  async handleUIBranch(uiFolder) {
+    let branch = GitUtils.DEFAULT_BRANCH;
+    try {
+      branch = await GitUtils.getBranch(uiFolder);
+
+      if (branch !== this._uiBranch) {
+        this.log.info(chalk`AEM Importer UI was on branch {yellow ${branch}}. Switching to {green ${this._uiBranch}}.`);
+        // This checkout will produce an error if the uiBranch does not exist (or fails to switch),
+        // will announce the error to the user, and exit the execution.
+        // Error examples:
+        // - "Your local changes to the following files would be overwritten by checkout: <file>"
+        // - "Could not find bad-branch."
```

Comment:
but the `git.checkout()` is caught...

Reply:
the comment above does not make sense.... is it a TODO ?

Reply:
Took the whole try/catch away and let the exception flow.  I think that simplified it a lot.